--- Makefile	2015-02-09 19:46:21.000000000 +0200
+++ Makefile-fixed	2015-02-25 14:28:43.087478387 +0200
@@ -1,7 +1,6 @@
-LDFLAGS=-L/usr/local/lib
-CFLAGS=-O3 -Wall -Wuninitialized -fomit-frame-pointer -funroll-loops \
-	-fstrength-reduce -DNODEBUG `libpng-config --I_opts`
-LDLIBS=-lpng -lz -lm
+CC=gcc
+override LDFLAGS+=$(shell pkg-config --libs libpng) -lz -lm
+override CFLAGS+=-DNODEBUG $(shell pkg-config --cflags libpng)
 
 all: optar unoptar
 
@@ -19,31 +18,31 @@
 	rm -f optar unoptar golay golay_codes.c *.o
 
 common.o: common.c optar.h
-	gcc -c $(CPPFLAGS) $(CFLAGS) -o $@ $<
+	$(CC) -c $(CFLAGS) -o $@ $<
 
 parity.o: parity.c
-	gcc -c $(CPPFLAGS) $(CFLAGS) -o $@ $<
+	$(CC) -c $(CFLAGS) -o $@ $<
 
 optar.o: optar.c optar.h font.h parity.h
-	gcc -c $(CPPFLAGS) $(CFLAGS) -o $@ $<
+	$(CC) -c $(CFLAGS) -o $@ $<
 
 golay_codes.o: golay_codes.c
-	gcc -c $(CPPFLAGS) $(CFLAGS) -o $@ $<
+	$(CC) -c $(CFLAGS) -o $@ $<
 
 golay.o: golay.c parity.h
-	gcc -c $(CPPFLAGS) $(CFLAGS) -o $@ $<
+	$(CC) -c $(CFLAGS) -o $@ $<
 
 unoptar.o: unoptar.c optar.h parity.h
-	gcc -c -I/usr/local/include/libpng $(CPPFLAGS) $(CFLAGS) -o $@ $<
+	$(CC) -c $(CFLAGS) -o $@ $<
 
 optar: optar.o common.o golay_codes.o parity.o
-	gcc $(LDFLAGS) -o $@ $^
+	$(CC) -o $@ $^ $(LDFLAGS)
 
 golay_codes.c: golay
 	./$< > $@
 
 golay: golay.o parity.o
-	gcc $(LDFLAGS) -o $@ $^ $(LDLIBS)
+	$(CC) -o $@ $^ $(LDFLAGS)
 
 unoptar: unoptar.o common.o golay_codes.o parity.o
-	gcc $(LDFLAGS) -o $@ $^ $(LDLIBS)
+	$(CC) -o $@ $^ $(LDFLAGS)
