diff --git a/Makefile b/Makefile
index 349ab7a..2e9fd43 100644
--- a/Makefile
+++ b/Makefile
@@ -17,14 +17,7 @@
 #   limitations under the License.
 #
 
-CC = gcc
-CFLAGS += -O2 -g -ggdb -c -std=c11 \
-	-D_GNU_SOURCE \
-	-fstack-protector-all -Wformat -Wformat=2 -Wformat-security -fPIE -D_FORTIFY_SOURCE=2 -Wa,--noexecstack \
-	-Wall -Wextra -Werror
-
-LD = gcc
-LDFLAGS += -Wl,-z,now -Wl,-z,relro -pie
+CFLAGS += -c -D_GNU_SOURCE -std=c11
 
 SRCS = nsjail.c cmdline.c contain.c log.c net.c subproc.c sandbox.c seccomp/bpf-helper.c
 OBJS = $(SRCS:.c=.o)
@@ -36,7 +29,7 @@ BIN = nsjail
 all: $(BIN)
 
 $(BIN): $(OBJS)
-	$(LD) -o $(BIN) $(OBJS) $(LDFLAGS)
+	$(CC) -o $(BIN) $(OBJS) $(LDFLAGS)
 
 clean:
 	$(RM) core Makefile.bak $(OBJS) $(BIN)
